clisops/core/regrid.py

```diff
@@ -0,0 +1,1854 @@
+
+    Returns
+    -------
+    None.
+    """
+    # Overwrite CONFIG entry with new value
+    CONFIG["clisops:grid_weights"]["local_weights_dir"] = weights_dir
+
+    # Create directory tree if required
+    if not os.path.isdir(weights_dir):
+        os.makedirs(weights_dir)
+
+
+# Initialize weights cache as defined in the clisops configuration (roocs.ini)
+weights_cache_init(CONFIG["clisops:grid_weights"]["local_weights_dir"])
+
+# Ensure local weight storage directory exists - decorator, used below
+check_weights_dir = functools.partial(
+    check_dir, dr=CONFIG["clisops:grid_weights"]["local_weights_dir"]
+)
+
+
+def weights_cache_flush(
+    weights_dir_init: str | Path | None = "",
+    dryrun: bool | None = False,
+    verbose: bool | None = False,
+):
+    """
+    Flush and reinitialize the local weights cache.
+
+    Parameters
+    ----------
+    weights_dir_init : str, optional
+        Directory name to reinitalize the local weights cache in.
+        Will be created if it does not exist.
+        The default is CONFIG["clisops:grid_weights"]["local_weights_dir"] as defined in roocs.ini
+        (or as redefined by a manual weights_cache_init call).
+    dryrun : bool, optional
+        If True, it will only print all files that would get deleted. The default is False.
+    verbose : bool, optional
+        If True, and dryrun is False, will print all files that are getting deleted.
+        The default is False.
+
+    Returns
+    -------
+    None.
+
+    """
+    # Read weights_dir from CONFIG
+    weights_dir = CONFIG["clisops:grid_weights"]["local_weights_dir"]
+
+    if dryrun:
+        print(f"Flushing the clisops weights cache ('{weights_dir}') would remove:")
+    elif verbose:
+        print(f"Flushing the clisops weights cache ('{weights_dir}'). Removing ...")
+
+    # Find and delete/report weight files, grid files and the json files containing the metadata
+    if os.path.isdir(weights_dir):
+        flist_weights = glob(f"{weights_dir}/weights_{'?'*32}_{'?'*32}_*.nc")
+        flist_meta = glob(f"{weights_dir}/weights_{'?'*32}_{'?'*32}_*.json")
+        flist_grids = glob(f"{weights_dir}/grid_{'?'*32}.nc")
+        if flist_weights != [] or flist_grids != [] or flist_meta != []:
+            for f in flist_meta + flist_weights + flist_grids:
+                if dryrun or verbose:
+                    print(f" - {f}")
+                if not dryrun:
+                    os.remove(f)
+        else:
+            if dryrun or verbose:
+                print("No weight or grid files found. Cache empty?")
+    elif dryrun:
+        print("No weight or grid files found. Cache empty?")
+
+    # Reinitialize local weights cache
+    if not dryrun:
+        if not weights_dir_init:
+            weights_dir_init = weights_dir
+        weights_cache_init(weights_dir_init)
+        if verbose:
+            print(f"Initialized new weights cache at {weights_dir_init}")
+
+
+class Grid:
```

Comment:
Since xESMF is a dependency, have you thought of subclassing the `xESMF.backend.Grid` instead ?
It only adds a `from_xarray` classmethod to the `ESMF.Grid` class. 

I could see it having a `from_grid_id` class method.